src/CompositeReferenceFieldManagerInterface.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\composite_reference;
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Field\FieldDefinitionInterface;
+
+/**
+ * Defines composite reference field managers.
+ */
+interface CompositeReferenceFieldManagerInterface {
+
+  /**
+   * Returns a list of possible entities that reference the given entity.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   The referenced entity.
+   *
+   * @return \Drupal\Core\Entity\EntityInterface[]
+   *   An array of entities that reference the given entity.
+   */
+  public function getReferencingEntities(EntityInterface $entity): array;
+
+  /**
+   * Reacts to the deletion of an entity and deletes its composite references.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   The referencing entity.
+   * @param \Drupal\Core\Field\FieldDefinitionInterface $field_definition
+   *   The definition of the field to remove references from.
+   */
+  public function onDelete(EntityInterface $entity, FieldDefinitionInterface $field_definition): void;
```

Comment:
I can call this service method anywhere, it does not really react to anything, you call it. It would be more accurate if you would call this method as deleteComposite() or deleteCompositeEntity() and you would say in the comment: "Deletes the composite references of a given entity."

In case this method is not meant to be called anywhere else, then please put the entity_delete() hook from the module file into a see tag and be more specific in the documentation.

Reply:
Let's rename it entityDelete(). This is how other core modules do it as well.